save-plugins/warn-plugin/src/commonTest/kotlin/org/cqfn/save/plugin/warn/sarif/SarifWarningAdapterTest.kt

```diff
@@ -0,0 +1,153 @@
+                        "uri": "file:///C:/dev/sarif/sarif-tutorials/samples/Introduction/simple-example.js"
+                      }
+                    }
+                  ],
+                  "results": [
+                    {
+                      "level": "error",
+                      "message": {
+                        "text": "'x' is assigned a value but never used."
+                      },
+                      "locations": [
+                        {
+                          "physicalLocation": {
+                            "artifactLocation": {
+                              "uri": "file:///C:/dev/sarif/sarif-tutorials/samples/Introduction/simple-example.js",
+                              "index": 0
+                            },
+                            "region": {
+                              "startLine": 1,
+                              "startColumn": 5
+                            }
+                          }
+                        }
+                      ],
+                      "ruleId": "no-unused-vars",
+                      "ruleIndex": 0
+                    }
+                  ]
+                }
+              ]
+            }
+        """.trimIndent()
+        val sarifSchema210: SarifSchema210 = Json.decodeFromString(sarif)
+
+        val warnings = sarifSchema210.toWarnings("C:/dev/sarif".toPath(), emptyList())
+
+        println(warnings)
+        assertEquals(1, warnings.size)
+        assertEquals(
+            Warning("'x' is assigned a value but never used.", 1, 5, "simple-example.js"),
+            warnings.first()
+        )
+    }
+
+    @Test
+    fun `should filter out warnings from other files`() {
+        val sarifSchema210 = SarifSchema210(
+            version = Version.The210,
+            runs = listOf(
+                runOf(uri = "file:///workspace/tests/suite1/foo.test"),
+                runOf(uri = "file:///workspace/tests/suite1/bar.test"),
+                runOf(uri = "file:///workspace/tests/suite2/foo.test"),
+            )
+        )
+
+        val testRoot = "/workspace/tests".toPath()
+        val warnings = sarifSchema210.toWarnings(
+            testRoot,
+            listOf("/workspace/tests/suite2/foo.test".toPath()).adjustToCommonRoot(testRoot)
+        )
+
+        println(warnings)
+        assertEquals(1, warnings.size)
+    }
+
+    @Test
+    fun `should filter out warnings from other files - relative paths`() {
+        val sarifSchema210 = SarifSchema210(
+            version = Version.The210,
+            runs = listOf(
+                runOf(uri = "suite1/foo.test"),
+                runOf(uri = "suite1/bar.test"),
+                runOf(uri = "suite2/foo.test"),
+            )
+        )
+
+        val testRoot = "/workspace/tests".toPath()
+        val warnings = sarifSchema210.toWarnings(
+            testRoot,
+            listOf("/workspace/tests/suite2/foo.test".toPath()).adjustToCommonRoot(testRoot)
+        )
+
+        println(warnings)
```

Comment:
debug?

Reply:
It's a test, and we use prints in tests in other places. I'll enable detekt's `ForbiddenMethodCall` for main code, though, it'll be helpful.